### PR TITLE
feat: add typst

### PR DESCRIPTION
**Pull request by sioodmy**
adds support for typst language [typst](https://typst.app/)

### CODE REVIEW

**Comment by SoraTenshi** [comment]
Thank you for the contrib.

Is this pr based on some original pr?
Because i have no way to validate the ts queries, as i have never used typst :D

File: runtime/queries/typst/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,77 @@
+(call
+  item: (ident) @function)
+(call
+  item: (field field: (ident) @function.method))
+(tagged field: (ident) @tag)
+(field field: (ident) @tag)
+(comment) @comment
+
+; CONTROL
+(let "let" @keyword.storage.type)
```
Comment (SoraTenshi) [inline runtime/queries/typst/highlights.scm:10]: having `(let "let"`... looks weird, is this really necessary?
i would assume, that the `let` node already captures it.
The same goes with all the others.

**Comment by sioodmy** [comment]
> Thank you for the contrib.
> 
> Is this pr based on some original pr? Because i have no way to validate the ts queries, as i have never used typst :D

It's backported from [this commit](https://github.com/helix-editor/helix/commit/290666011971c79e36b5172eea66b315dc52e2c9)

I can confirm its fully functional

**Comment by sioodmy** [comment]
image

**Comment by SoraTenshi** [comment]
THX! :)

**Pull request merged**